src/video.js

```diff
@@ -64,3 +66,47 @@ export const checkVideoBidSetup = hook('sync', function(bid, adUnit, videoMediaT
 
   return true;
 }, 'checkVideoBidSetup');
+
+export function registerVASTTrackers(tracker) {
+  vastTrackers.push(tracker);
+}
+
+export function insertVastTrackers(trackers, vastXml) {
+  const doc = new DOMParser().parseFromString(vastXml, 'text/xml');
+  const wrappers = doc.querySelectorAll('VAST Ad Wrapper, VAST Ad InLine');
+  try {
+    if (wrappers.length) {
+      wrappers.forEach(wrapper => {
+        trackers['impressions'].forEach(trackingUrl => {
+          const impression = doc.createElement('Impression');
+          impression.appendChild(doc.createCDATASection(trackingUrl));
+          wrapper.appendChild(impression)
+        });
+      });
+      vastXml = new XMLSerializer().serializeToString(doc);
+    }
+  } catch (error) {
+    logError('an error happened trying to insert trackers in vastXml');
+  }
+  return vastXml;
+}
+
+export function getVastTrackers(bid) {
+  let hasTrackers = false;
+  let trackers = {'impressions': []};
+  vastTrackers.forEach(func => {
```

Comment:
what are the elements in the `vastTrackers` array ? They are functions that when given a bid as an arg return what ?

Reply:
here is an example of implementation at the analytic module side :
```
registerVASTTrackers(function(bidResponse) {
      return {
        'impressions': [`https://tracking.mydomain.com/vast?cpm=${bidResponse.cpm}`]
      };
    });
```
The function registered in `vastTrackers` returns an object with as many tracking url needed for each events. Currently only impression is handled, but doing it this way make it possible to extend to other vast tracking such as start, quartiles, completion...

Reply:
It's also the reason why `getVastTrackers` returns the kind of structure its `trackers` variable is.